enum.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    if block_given?
+      while i < size
+        yield(self[i])
+        i += 1
+      end
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    if block_given?
+      while i < size
+        yield(self[i], i)
+        i += 1
+      end
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    i = 0
+    new_array = []
+    while i < size
+      new_array << self[i] if yield(self[i]) == true
+      i += 1
+    end
+    new_array
+  end
+
+  def my_all?(arg = nil)
+    if block_given?
+      my_each { |x| return false unless yield(x) }
+    elsif arg.class == Class
+      my_each { |x| return false unless x.class == arg }
+    elsif arg.class == Regexp
+      my_each { |x| return false if (x =~ arg).nil? }
+    elsif !arg.nil?
+      my_each { |x| return false unless x == arg }
+    elsif arg.nil?
+      my_each { |x| return false unless x }
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
```

Comment:
Your `my_any` does not quite work as should.
Two considerations:
- When no block or argument is given it should return `true` if at least one of the elements in the array is `true`. Otherwise, it returns `false`. Line 63 does not address this as you think. [See comment]
- when a pattern other than Regex or a Class is given it should `false` if none of the collection matches the pattern and `true` otherwise

You should consider returning false by default and return true if an element meets criteria. This would solve a lot of your problems